src/dialect.rs

```diff
@@ -0,0 +1,55 @@
+use crate::allocator::{Allocator, NodePtr};
+use crate::cost::Cost;
+use crate::reduction::Response;
+use crate::run_program::{OperatorHandler, PreEval, RunProgramContext};
+
+pub struct Dialect {
+    quote_kw: Vec<u8>,
+    apply_kw: Vec<u8>,
+    op_handler: Box<dyn OperatorHandler>,
+}
+
+impl OperatorHandler for Dialect {
+    fn op(
+        &self,
+        allocator: &mut Allocator,
+        op: NodePtr,
+        args: NodePtr,
+        max_cost: Cost,
+    ) -> Response {
+        self.op_handler.op(allocator, op, args, max_cost)
```

Comment:
instead of forwarding this call, could the `Dialect` trait and `OperatorHandler` traits just be merged? They seem simple enough and related enough that it could just be one `Dialect` trait